### PR TITLE
Add drain upgrade guide docs

### PR DESCRIPTION
**Pull request by schmichael**
Depends on #4142 and #4144

### CODE REVIEW

File: website/source/docs/upgrade/upgrade-specific.html.md [hunk 1]
```diff
@@ -51,6 +51,16 @@ Raft Protocol versions supported by each Consul version:
 In order to enable all [Autopilot](/guides/cluster/autopilot.html) features, all servers
 in a Nomad cluster must be running with Raft protocol version 3 or later.
 
+### Node Draining Improvements
+
+Node draining via the [`node drain`][drain-cli] command or the [drain
+API][drain-api] has been substantially changed in Nomad 0.8. In Nomad 0.7.1 and
+earlier draining a node would immediately stop all allocations on the node
+being drained. Nomad 0.8 now supports a [`migrate`][migrate] stanza in job
```
Comment (chelseakomlo) [inline website/source/docs/upgrade/upgrade-specific.html.md]: Is the default draining behavior in Nomad 0.8 the same without the `migrate` stanza? If not, specify that here.

File: website/source/docs/upgrade/upgrade-specific.html.md [hunk 2]
```diff
@@ -51,6 +51,16 @@ Raft Protocol versions supported by each Consul version:
...
+specifications to control how many allocations may be migrated at once.
```
Comment (chelseakomlo) [inline website/source/docs/upgrade/upgrade-specific.html.md]: Maybe specify the advantage of this unless the decommissioning nodes guides does so (i.e, to ensure a service always is alive even as nodes are being decommissioned).

File: website/source/docs/upgrade/upgrade-specific.html.md (hunk 1 above)
Comment (schmichael) [reply website/source/docs/upgrade/upgrade-specific.html.md]: Good call; added.

**Comment by dadgar** [comment]
Hm I am not sure we want to add this PR. The upgrade specific section is for things that an operator should be aware of when upgrading (as in extra steps or precautions). This just documents the feature.

I do think we should add a link to the guide here: https://www.nomadproject.io/docs/upgrade/index.html#5-upgrade-clients

**Pull request merged**